### ISSUE TITLE
`npx watch` in dev script stops executing if file references `googleapis`

### ISSUE THREAD

**Issue by osztenkurden** [vercel/turbo#7834]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/osztenkurden/turbo-tsx-googleapis-issue

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Windows

### Which canary version will you have in your reproduction?

1.13.1-canary.0

### Describe the Bug

Running turbo command, that invokes `tsx watch FILE.ts`, when FILE.ts includs reference from `googleapis` npm package results in no output, and no code being run. Removing `watch` flag makes it work properly. This worked properly on 1.12.5, and breaks on 1.13+.

### Expected Behavior

Referencing `googleapis` in Typescript code should not freeze executing that file when run with `tsx watch`

### To Reproduce

apps/backend/index.ts:
```typescript
import { google } from 'googleapis';
console.log({ google })
```

apps/backend/package.json:
```json
{
  "name": "api",
  "scripts": {
    "dev": "tsx watch src/index.ts",
    "build": "npx tsc"
  },
  "dependencies": {
    "googleapis": "^134.0.0"
  },
  "devDependencies": {
    "tsx": "^4.7.1",
    "typescript": "^4.9.4"
  }
}
```

./turbo.json:
```json
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["prebuild", "^build"],
      "outputs": ["output-file.txt", "dist/**"]
    },
    "prebuild": {},
    "lint": {},
    "type-check": {},
    "dev": {
      "cache": false
    }
  }
}
```

./package.json:
```json
{
  "name": "my-turborepo",
  "description": "A barebones Turborepo example for working with Task Graphs.",
  "devDependencies": {
    "turbo": "canary"
  },
  "scripts": {
    "dev": "turbo run dev"
  },
  "packageManager": "npm@10.1.0",
  "workspaces": [
    "apps/*",
    "packages/*"
  ]
}
```

Running `npm run dev` should result in logging the content of the `google` object from `googleapis`. Currently it hangs:
![image](https://github.com/vercel/turbo/assets/25458209/8d99e135-b0bc-45e0-9c4f-68d79e35260c)

This works fine in 1.12.5, as well when manually executing `npx tsc watch src/index.ts`

**Comment by NicholasLYang** [comment]
Hi @osztenkurden, could you try adding [`"persistent": true`](https://turbo.build/repo/docs/reference/configuration#persistent) to your `dev` task? Since it's a long running task, it needs that option to run correctly. Let me know if that helps!

**Comment by osztenkurden** [comment]
@NicholasLYang Updated turbo.json so it looks like that:
```json
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["prebuild", "^build"],
      "outputs": ["output-file.txt", "dist/**"]
    },
    "prebuild": {},
    "lint": {},
    "type-check": {},
    "dev": {
      "cache": false,
      "persistent": true
    }
  }
}
```
However, still freezes.

**Comment by chris-olszewski** [comment]
Can you confirm that the script works without the use of `turbo`?

I just tried cloning your repo and `cd apps/backend && npm run dev` resulted in the same behavior without use of `turbo`:
![image-5](https://github.com/vercel/turbo/assets/4131117/345c74a3-57e6-41a0-9744-27c7288a1f6f)

**Comment by osztenkurden** [comment]
@chris-olszewski Yes, it does work fine without using turbo. Running just `npx tsx watch src/index.ts` or `tsx watch src/index.ts` both work

**Comment by osztenkurden** [comment]
@chris-olszewski Yep, it does, to be sure I verified both Terminal/CMD and Git Bash, both console log out:
![image](https://github.com/vercel/turbo/assets/25458209/beb2c6e9-c59c-4fcc-bff4-7b5c82d7e4a7)
![image](https://github.com/vercel/turbo/assets/25458209/43acd962-28a0-4229-9843-2e29c97f8a00)
Thats very weird, because it only starts happening on Turbo 13

**Comment by chris-olszewski** [comment]
Okay, I've been able to track this down to a weird interaction with Windows, `tsx watch`, with tasks that aren't hooked up to a PTY. 

Not exactly sure what the fix is at this time, but one workaround is to use the new [experimental UI](https://turbo.build/blog/turbo-1-13-0#new-terminal-ui) which allows for tasks on Windows to be hooked up to a pseudoterminal.

**Comment by anthonyshew** [comment]
Hey, folks. From everything I can tell this issue appears to be resolved. I'm unable to reproduce it with today's Turborepo. I've also added a regression test for this behavior in #12609.
